### DISCUSSION TITLE
Do Edgex have its own API gateway so that Edgex APIs could be managed?

### DISCUSSION THREAD

**Discussion by pachori91** [edgexfoundry/community#212] [Q&A]
Hi Team,

Is there any API Gateway of edgex where different apis could be managed? If Edgex have api gateway what are the features it provides?
Actually i am looking for answer for below queries-

- Can we apply device level security through gateway?
- Can we manage access of different apis for users?
- Can we manage access on different device data?
- Can we find usage of apis?

Thanks

**Comment by cloudxxx8** [comment]
Yes, we have nginx as the default API Gateway
https://docs.edgexfoundry.org/3.1/security/Ch-APIGateway/

You can modify the nginx config to fulfill your requirement.

**Comment by jiekechoo** [comment]
@pachori91 
Maybe you need a enterprise system or application for manage EdgeX, and `the API Gateway` is setting on EdgeX.

Jieke
YIQISOFT

**Comment by bnevis-i** [comment]
> If Edgex have api gateway what are the features it provides?

Note there is a related discussion here <https://github.com/orgs/edgexfoundry/discussions/213> that is inquiring about RBAC for EdgeX.  There are no current documented use cases for an authorization model other than all-or-nothing.  It would be beneficial to the community and perhaps push your use case forward if you were to propose one.
